smf/nexus/config-partial.toml

```diff
@@ -38,3 +38,8 @@ dns_external.max_concurrent_server_updates = 5
 # certificates it will take _other_ Nexus instances to notice and stop serving
 # them (on a sunny day).
 external_endpoints.period_secs = 60
+
+[default_region_allocation_strategy]
+# by default, allocate across 3 distinct sleds
+# seed is omitted so a new seed will be chosen with every allocation.
+type = "random_with_distinct_sleds"
```

Comment:
So I chatted with @iliana about this earlier, and here's the TL;DR of our conversation:

- Most of the "non-deploy" tests work using the config from `nexus/tests/config.test.toml`
- The "real prod config" is using this file (`smf/nexus/config-partial.toml`), and uses the distinct sled strategy, as we want

So the problem cases are the following:
- Developers trying to run a single sled
- The "deploy job", which typically runs a "non-gimlet" standalone sled (similar to the developer workflow)

These cases are painful, because we want **some** config options to be like prod (e.g., we use internal DNS! We have an underlay, kinda!) but others we **don't** want to look like our prod environment (most obviously: we're only running with a single sled).

iliana and I came to the conclusion that a couple ways to solve this would be the following:

## Add a new target option to `omicron-package` (my slight preference)

1. Add a new feature to the omicron-package target type named `cluster`, which can be either `single-sled` or `multi-sled`. This would involve updating `package/src/lib.rs` and `package/src/target.rs`. There is **some** overlap with the `gimlet` vs `gimlet-standalone` vs `non-gimlet` options, but I think it's probably worth just adding a new feature explicitly.
2. Split the Nexus config files into two: what previously was `smf/nexus/config-partial.toml` becomes `smf/nexus/multi-sled/config-partial.toml` and `smf/nexus/single-sled/config-partial.toml`. We can provide a different region allocation strategy in each, through the `PackageConfig`.
3. Update `package-manifest.toml` to reference the right `smf` subdirectory for Nexus, based on the `cluster` value (you can look at how we do this for the `sled-agent` target using the `machine` target).

Pros: Plumbs Nexus-specific options directly to Nexus
Cons: Possible to select confusing combinations of targets, like "gimlet-standalone" + "multi-sled"?

## Try to overload an existing target?

1. Use the `non-gimlet` and `gimlet-standalone` options, and "infer" that these are single-sled
2. Pass an argument from sled agent to Nexus (via `DeploymentConfig`) when this is the case

Pros: One fewer target?
Cons: Prevents us from effectively having a multi-sled setup without real gimlets (I think this would break the Canada cluster), kinda confusing that a Nexus-level option is plumbed through the sled agent config

Reply:
you've sold me on adding a new target option to omicron-package i think.

Reply:
What `cluster` option will I want to pass if I want to set up a multi-sled dev cluster with exactly two sleds? It seems like the `single-sled` option should still work for now (though it makes me wonder if there's a clearer label we could put on it); is that correct?